agent/eventhandler/task_handler_types.go

```diff
@@ -28,9 +29,13 @@ type sendableEvent struct {
 
 	taskSent   bool
 	taskChange api.TaskStateChange
+
+	mu sync.RWMutex
```

Comment:
This lock is protecting the [`*Sent`](https://github.com/aws/amazon-ecs-agent/pull/857/files#diff-7ef9683a9c7955f55d54ba8d4f4f622bR99) bools here, this is really protecting mutation of the entire struct, and now the reading of the `isContainerEvent` field. The `mu` name does make sense as it stands IMO.